src/group_theory/perm/sign.lean

```diff
@@ -693,4 +714,128 @@ using_well_founded {rel_tac := λ _ _, `[exact ⟨_, measure_wf (λ f, f.support
 
 end sign
 
+lemma swap_eq_prod_swap_adj (n : ℕ) {i j : fin n} (hij : i ≠ j) : ∃ L : list (perm (fin n)),
+L.prod = equiv.swap i j ∧ ∀ l ∈ L, ∃ k : fin n, ∃ h : k.1 + 1 < n,
+ l = equiv.swap k ⟨k.1 + 1, h⟩ :=
```

Comment:
This is quite a complicated statement. Would it be easier to say that there a list of nats, such that `equiv.swap i j` is the product of the corresponding adjacent transpositions?

Reply:
Also, it might be easier if you split this into several lemmas: prove the first one with the hypothesis that `i < j`.  You could be very explicit about what the sequence of adjacent transpositions is, rather than just an existential statement.

Reply:
If I were to state the lemma using a list of nats, I can't think of a good way of keeping track of an `l.1 + 1 < n` for each `l ∈ L`.

Reply:
You could, e.g. have `i j : fin (n+1)`, then produce in one declaration a `list (fin n)`, and then in a second declaration `equiv.swap i j = (L.map ...).prod`, where the ... converts a `fin n` into the adjacent transposition of `fin (n+1)`.

I'm not saying this is essential, just potentially results in cleaner statements, and proofs split into smaller chunks.